test/distributed/_composable/fsdp/test_fully_shard_training.py

```diff
@@ -797,11 +788,9 @@ def _test_gradient_accumulation(
                 expected_all_gather_count += num_mlps * (num_microbatches - 1)
             self.assertEqual(all_gather_count, expected_all_gather_count)
 
-            # Average the ref model's gradients over the world size to match
-            # data parallel semantics
             for param in ref_model.parameters():
                 if param.grad is not None:
-                    param.grad.div_(self.world_size)
+                    dist.all_reduce(param.grad, op=dist.ReduceOp.AVG)
```

Comment:
iirc the ref model is a dp model now, whereras the previous approach is more like a `single device` model?

This would incur additional allreduce communications compare to the single device approach though 🤔

Reply:
It does incur extra all-reduce, but that all-reduce is outside the `CommDebugMode` region (so it does not affect the counts).